game/scripts/vscripts/TutorialGraph/Steps.ts

```diff
@@ -139,7 +150,10 @@ export const moveUnit = (unit: tg.StepArgument<CDOTA_BaseNPC>, moveLocation: tg.
             Queue: true
         }
 
-        ExecuteOrderFromTable(order)
+        if (actualUnit.IsAlive())
+            ExecuteOrderFromTable(order)
+        else
+            complete() // Unit is dead, skip doing anything
```

Comment:
I think we sohuld check if the unit is alive in `checkIsIdle` too since the unit could die while it's moving. Also I'm not sure,  maybe we should call error instead of completing the step (or do you need it to work when the unit dies too? It'd be considered a bug if that happened in most places I think.).

Reply:
Wasn't sure about adding this, in my case I wouldn't call it a bug since a unit could possibly die and I want to move order them later if they are alive. I could wrap a completeCheck step and moveUnit step with a forkAny and use that as a way to prevent that happening, and revert this change?

Reply:
Yea I think so, although that might cause an error if the stuff above is called before your other step. Maybe the function should take an optional boolean for whether we want to complete when the unit is invalid (`!isUnitValidAndAlive(unit)`) with default `false`.